pyro/infer/reparam/stable.py

```diff
@@ -99,11 +99,101 @@ def __call__(self, name, fn, obs):
+        When validation is enabled this will raise a ``ValueError`` if
+        stability is close to 1.
+
+    [1] V. M. Zolotarev (1986)
+        "One-dimensional stable distributions"
+    """
+
+    def __call__(self, name, fn, obs):
+        fn, event_dim = self._unwrap(fn)
+        assert isinstance(fn, dist.Stable)
+        if is_validation_enabled():
+            if (fn.stability - 1).abs().lt(0.01).any():
+                raise ValueError("StableReparam found stability near 1")
+
+        # Strategy: Let X ~ S0(a,b,s,m) be the stable variable of interest.
+        # 1. WLOG scale and shift so s=1 and m=0, additionally shifting to convert
+        #    from Zolotarev's S parameterization to Nolan's S0 parameterization.
+        # 2. Decompose X = S + T, where
+        #    S ~ S(a,0,...,0) is symmetric and
+        #    T ~ S(a,sgn(b),...,0) is totally skewed.
+        # 3. Decompose S = G * sqrt(Z) via the symmetric strategy, where
+        #    Z ~ S(a/2,1,...,0) is totally-skewed and
+        #    G ~ Normal(0,1) is Gaussian.
+        # 4. Defer the totally-skewed Z and T to the Chambers-Mallows-Stuck
+        #    strategy: Z = f(Unif,Exp), T = f(Unif,Exp).
+        #
+        # To derive the parameters of S and T, we solve the equations
+        #
+        #   T.stability = a            S.stability = a
+        #   T.skew = sgn(b)            S.skew = 0
+        #   T.loc = 0                  S.loc = 0
+        #
+        #   s = (S.scale**a + T.scale**a)**(1/a) = 1       # by step 1.
+        #
+        #       S.skew * S.scale**a + T.skew * T.scale**a
+        #   b = ----------------------------------------- = sgn(b) * T.scale**a
+        #                S.scale**a + T.scale**a
+        # yielding
+        #
+        #   T.scale = |b| ** (1/a)     S.scale = (1 - |b|) ** (1/a)
+
+        # Draw parameter-free noise.
+        proto = fn.stability
+        half_pi = proto.new_full(proto.shape, math.pi / 2)
+        one = proto.new_ones(proto.shape)
+        zu = pyro.sample("{}_z_uniform".format(name),
+                         self._wrap(dist.Uniform(-half_pi, half_pi), event_dim))
+        ze = pyro.sample("{}_z_exponential".format(name),
+                         self._wrap(dist.Exponential(one), event_dim))
+        tu = pyro.sample("{}_t_uniform".format(name),
+                         self._wrap(dist.Uniform(-half_pi, half_pi), event_dim))
+        te = pyro.sample("{}_t_exponential".format(name),
+                         self._wrap(dist.Exponential(one), event_dim))
+
+        # Differentiably transform.
+        a = fn.stability
+        z = _unsafe_standard_stable(a / 2, 1, zu, ze, coords="S")
+        t = _standard_stable(a, one, tu, te, coords="S")
+        a_inv = a.reciprocal()
+        skew_abs = fn.skew.abs()
+        t_scale = skew_abs.pow(a_inv) * fn.skew.sign()
+        s_scale = (1 - skew_abs).pow(a_inv)
+        # FIXME The following line is discontinuous at a=1.
```

Comment:
@fehiepsi I don't see an easy way to fix this, and as @martinjankowiak points out most interesting datasets have `stability > 1.5`. Therefore I've simply added a FIXME, a warning in the docstring, and a validation check. We can think about this in the future if we ever need `stability` near 1.

Ready to merge, pending review.

Reply:
Oh wait, I think I see the fix :smile: The following function **is** continuous except at the hole
```
(t_scale - fn.skew) * tan(pi / 2 * a)
  = skew.sign() * (skew.abs().pow(1/a) - skew.abs()) * tan(pi / 2 * a)
                  ----------------------------------   ---------------
                             has a root                has an asymptote
```
so we simply need to work around the canceling root. I'll quick add that as another interpolation trick...

Reply:
Fixed :smile: Ready to merge, pending review.

Reply:
Brilliant!

I spent the whole night to figure out a solution but unsuccessful. Pretty happy when seeing your solution after waking up! :D

Reply:
Ha ha I also spent the whole night looking for a solution :D